02-lb-git.md

```diff
@@ -0,0 +1,145 @@
+Among other features, Gitlab allows you to browse the source code for each project,
+and to review new changes (called *merge requests*) in a convenient web interface.
+You can find the CERN Gitlab instance at [https://gitlab.cern.ch](https://gitlab.cern.ch).
+
+In principle, there are multiple ways of interacting with the LHCb software repositories:
+
+ 1. A vanilla git workflow using only the standard git commands.
+   This requires you to `clone` and compile an entire LHCb project at a time.
+ 2. An LHCb-specific workflow using a set of `lb-*` subcommands.
+    This allows you to check out individual packages inside a project, and
+    streamlines the modification of a few packages at a time.
+    This is closer to the previously used `getpack` command.
+
+Here, we want to focus on the second workflow.
+
+Before jumping in by creating a project in Gitlab, you should make sure that
+your local git configuration and your settings on Gitlab are sufficiently set up:
+
+Your name and email address should be set up in your local `git` configuration.
+To ensure that this is the case, run
+```
+$ git config --global user.name "Your Name"
+$ git config --global user.email "Your Name <your.name@cern.ch>"
+```
+and put in your information.
+
+Next, connect to [https://gitlab.cern.ch](https://gitlab.cern.ch) and log in
+with your CERN credentials.
+
+Visit [https://gitlab.cern.ch/profile/keys](https://gitlab.cern.ch/profile/keys) and add an SSH key.
+
+Run this LHCb-specific configuration command:
+```
+git config --global lb-use.protocol ssh
+```
+This makes sure the LHCb commands use the ssh protocol instead of https.
+
+In the Gitlab web interface, create a new project by clicking on the "New project" button.
+Give your project the name "LHCbSK".
+
+We will now set up a local LHCb development area that is connected to the git
+repository you just created.
+In order to create a new dev environment, run
+```
+$ lb-dev --list LHCb
+```
+to see which versions of the LHCb project are available.
+Pick the newest one and run
+```
+$ lb-dev --name LHCbSK LHCb/<version>
+```
+where you need to insert the version you picked earlier.
+This will automatically run `git init` to create a new git repository with some initial files (`Makefile`, `CMakeLists.txt`, etc.).
+In order to connect your repository with the remote one, run
+```
+cd LHCbSK
+git remote add origin ssh://git@gitlab.cern.ch:7999/<username>/LHCbSK.git
+```
+where you need to substitute your username.
+Run
+```
+git push -u origin master
+```
+once to define `master` as the remote branch that we should push to by default.
+
+For the purpose of this tutorial, we've set up an `SKTest` project containing
+the package `TheTestPackage`.
+In order to define `SKTest` as your currently active project, run
+```
+git lb-use SKTest
+```
+This will also fetch the `SKTest` git repository from Gitlab.
+
+You can then run
+```
+git lb-checkout SKTest TheTestPackage
```

Comment:
I think you have to specify the branch, i.e.
git lb-checkout SKTest/master TheTestPackage